wasm/lib/src/lib.rs

```diff
@@ -47,14 +49,39 @@ pub fn setup_console_error() {
 #[wasm_bindgen(typescript_custom_section)]
 const TS_CMT_START: &'static str = "/*";
 
+fn run_py(source: &str, options: Option<Object>, mode: Mode) -> Result<JsValue, JsValue> {
+    let vm = VMStore::init(PY_EVAL_VM_ID.into(), Some(true));
+    let options = options.unwrap_or_else(Object::new);
+    let js_vars = {
+        let prop = Reflect::get(&options, &"vars".into())?;
+        if prop.is_undefined() {
+            None
+        } else if prop.is_object() {
+            Some(Object::from(prop))
+        } else {
+            return Err(TypeError::new("vars must be an object").into());
+        }
+    };
+
+    vm.set_stdout(Reflect::get(&options, &"stdout".into())?)?;
+
+    if let Some(js_vars) = js_vars {
+        vm.add_to_scope("js_vars".into(), js_vars.into())?;
+    }
```

Comment:
This match is basically what `VMStore.run` does -- you might want to make that `pub(crate)` and just call it here.